cmd/k8s-operator/hostscm.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !plan9
+
+// tailscale-operator provides a way to expose services running in a Kubernetes
+// cluster to your Tailnet.
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"tailscale.com/client/tailscale/apitype"
+)
+
+const dnsConfigKey = "dns.json"
+
+// hostsCMConfig contains the config needed to update ts.net config for a
+// particular egress proxy
+type hostsCMConfig struct {
+	// IP of the Tailscale service that we are setting up egress for
```

Comment:
Perhaps "Tailscale node" rather than "service"?

Reply:
Thanks, I was trying to think of a better name, as we already have a Kubernetes service!